--- ekg2-20050405/ekg/userlist.h.orig	2004-10-04 22:07:26.000000000 +0200
+++ ekg2-20050405/ekg/userlist.h	2005-05-04 22:48:01.000000000 +0200
@@ -116,13 +116,14 @@
 };
 
 #define	IGNORE_LABELS_MAX 7
-struct ignore_label ignore_labels[IGNORE_LABELS_MAX];
 
 struct ignore_label {
 	int level;
 	char *name;
 };
 
+struct ignore_label ignore_labels[IGNORE_LABELS_MAX];
+
 //list_t userlist;
 
 int userlist_read(session_t* session);
