--- ekg2-20060622/configure.ac~	2006-06-25 23:04:08.000000000 +0200
+++ ekg2-20060622/configure.ac	2006-06-25 23:25:02.000000000 +0200
@@ -364,7 +364,6 @@
 
 	if test "x$have_perl" = "xyes"; then
 		AC_CONFIG_FILES([plugins/perl/Makefile])
-		AC_CONFIG_FILES([contrib/perl/Makefile])
 		PLUGINS="$PLUGINS perl"
 	fi
 fi
@@ -378,7 +377,6 @@
 
 	if test "x$have_python" = "xyes"; then
 		AC_CONFIG_FILES([plugins/python/Makefile])
-		AC_CONFIG_FILES([contrib/python/Makefile])
 		PLUGINS="$PLUGINS python"
 	fi
 fi
