--- ekg2-20060831/configure.ac.fix	2006-09-01 07:46:20.644070250 +0200
+++ ekg2-20060831/configure.ac	2006-09-01 07:46:38.705199000 +0200
@@ -94,7 +94,7 @@
 AC_CHECK_FUNC(dlopen, [], [AC_CHECK_LIB(dl, dlopen, DLLIB="$DLLIB -ldl")])
 
 AC_CONFIG_FILES([Makefile ekg/Makefile ekg/ekg2-config compat/Makefile docs/Makefile po/Makefile.in])
-AC_CONFIG_FILES([contrib/Makefile contrib/python/Makefile contrib/perl/Makefile])
+AC_CONFIG_FILES([contrib/Makefile])
 AC_CONFIG_FILES([plugins/Makefile plugins/mail/Makefile plugins/logs/Makefile plugins/rc/Makefile plugins/sms/Makefile plugins/pcm/Makefile plugins/rot13/Makefile plugins/irc/Makefile plugins/httprc_xajax/Makefile plugins/xmsg/Makefile plugins/ioctld/Makefile plugins/gg/Makefile plugins/sim/Makefile plugins/jabber/Makefile plugins/ncurses/Makefile plugins/gtk/Makefile plugins/readline/Makefile plugins/gsm/Makefile plugins/xosd/Makefile plugins/perl/Makefile plugins/python/Makefile plugins/logsqlite/Makefile plugins/gpg/Makefile])
 
 PLUGINS="logs mail rc sms pcm irc rot13 httprc_xajax xmsg"
@@ -410,7 +410,6 @@
 	AM_CHECK_PERL
 
 	if test "x$have_perl" = "xyes"; then
-		CONTRIB_SUBDIRS="$CONTRIB_SUBDIRS perl"
 		PLUGINS="$PLUGINS perl"
 	fi
 fi
@@ -421,7 +420,6 @@
 	AM_CHECK_PYTHON
 
 	if test "x$have_python" = "xyes"; then
-		CONTRIB_SUBDIRS="$CONTRIB_SUBDIRS python"
 		PLUGINS="$PLUGINS python"
 	fi
 fi
